File: shop-admin-backend/src/main/java/com/demo/admin/core/models/BaseModel.java
package com.demo.admin.core.models;

import jakarta.persistence.Column;
import jakarta.persistence.MappedSuperclass;
import lombok.*;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.sql.Timestamp;

@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@EqualsAndHashCode(callSuper = false)
@MappedSuperclass
public class BaseModel {

    @Column(name = "created_by")
    private Integer createdBy;

    @Column(name = "deleted_by")
    private Integer deletedBy;

    @Column(name = "updated_by")
    private Integer updatedBy;

    @Column(name = "is_active")
    private Boolean isActive;

    @Column(name = "metadata", columnDefinition = "json")
    private String metadata;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false, insertable = false)
    private Timestamp createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", updatable = false, insertable = false)
    private Timestamp updatedAt;


    @Column(name = "deleted_at")
    private Timestamp deletedAt;
}

File: shop-backend/src/@core/interfaces/models/base-model.ts
import { Expose } from 'class-transformer'
import { IsISO8601 } from 'class-validator'

export class BaseModel {
	@Expose()
	metadata?: object

	@Expose()
	is_active!: boolean

	@Expose()
	created_at!: Date

	@Expose()
	created_by?: number

	@Expose()
	deleted_at?: Date | null

	@Expose()
	deleted_by?: number

	@Expose()
	updated_at?: Date | null

	@Expose()
	updated_by?: number
}
